pkg/cmd/unpack/unpack.go

```diff
@@ -77,32 +77,52 @@ func runUnpackRecursive(ctx context.Context, opts *unpackOptions, visitedRefs []
 
 	// Since there might be multiple datasets, etc. we need to synchronously iterate
 	// through the config's relevant field to get the correct path for unpacking
+	// We need to support older ModelKits (that were packed without diffIDs and digest
+	// in the config) for now, so we need to continue using the old structure.
 	var modelPartIdx, codeIdx, datasetIdx, docsIdx int
 	for _, layerDesc := range manifest.Layers {
+		// This variable supports older-format tar layers (that don't include the
+		// layer path). For current ModelKits, this will be empty
 		var relPath string
+
 		mediaType := constants.ParseMediaType(layerDesc.MediaType)
 		switch mediaType.BaseType {
 		case constants.ModelType:
 			if !shouldUnpackLayer(config.Model, opts.filterConfs) {
 				continue
 			}
-			_, relPath, err = filesystem.VerifySubpath(opts.unpackDir, config.Model.Path)
-			if err != nil {
-				return fmt.Errorf("error resolving model path: %w", err)
+			if config.Model.LayerInfo != nil {
```

Comment:
Should this also check if there is a `config.Model` ?

Reply:
We only enter this branch of the switch if we encounter a model layer, which implies the existence of a `model` field in the Kitfile (i.e. how did we pack a model layer without a model section in the Kitfile?).